subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/mqtt/MqttNotificationServiceTest.kt

```diff
@@ -0,0 +1,176 @@
+package com.egm.stellio.subscription.service.mqtt
+
+import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_SUBSCRIPTION_TERM
+import com.egm.stellio.shared.util.toUri
+import com.egm.stellio.subscription.model.*
+import com.egm.stellio.subscription.support.WithMosquittoContainer
+import com.ninjasquad.springmockk.SpykBean
+import io.mockk.coEvery
+import io.mockk.coVerify
+import kotlinx.coroutines.test.runTest
+import org.eclipse.paho.client.mqttv3.MqttException
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import java.net.URI
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE, classes = [MqttNotificationService::class])
+@ActiveProfiles("test")
+class MqttNotificationServiceTest : WithMosquittoContainer {
```

Comment:
one thing just came up into my mind: would be nice to have a test where you check you have actually received the notification through MQTT.